cmd/ausoceantv/auth.go

```diff
@@ -44,7 +47,33 @@ func (svc *service) logoutHandler(c *fiber.Ctx) error {
 
 // callbackHandler handles callbacks from google's oauth2 flow.
 func (svc *service) callbackHandler(c *fiber.Ctx) error {
-	return svc.CallbackHandler(c)
+	err := svc.CallbackHandler(c)
+	if err != nil {
+		return err
+	}
+
+	// Check if a user already exists.
+	p, err := svc.GetProfile(c)
+	if errors.Is(err, SessionNotFound) || errors.Is(err, TokenNotFound) {
```

Comment:
Would it be useful to have separate cases for these and then you can wrap the fiber error with more information i.e. whether it was a ```SessionNotFound``` or a ```TokenNotFound``` ?

Also, why exactly are we using the fiber error ?